extensions/notebook/src/dialog/managePackages/addNewPackageTab.ts

```diff
@@ -0,0 +1,264 @@
+				label: this.InstallButtonText,
+				enabled: false,
+				width: '80px'
+			}).component();
+			this.packageInstallButton.onDidClick(() => {
+				this.doPackageInstall();
+			});
+
+			let formModel = view.modelBuilder.formContainer()
+				.withFormItems([{
+					component: this.newPackagesSearchBar,
+					title: ''
+				}, {
+					component: this.packagesSearchButton,
+					title: ''
+				}, {
+					component: this.newPackagesNameLoader,
+					title: this.PackageNameTitle
+				}, {
+					component: this.newPackagesVersionsLoader,
+					title: this.PackageVersionTitle
+				}, {
+					component: this.newPackagesSummaryLoader,
+					title: this.PackageSummaryTitle
+				}, {
+					component: this.packageInstallButton,
+					title: ''
+				}]).component();
+
+			await view.initializeModel(formModel);
+		});
+	}
+
+	public get tab(): azdata.window.DialogTab {
+		return this.addNewPkgTab;
+	}
+
+	private async toggleNewPackagesFields(enable: boolean): Promise<void> {
+		await this.packagesSearchButton.updateProperties({ enabled: enable });
+		await this.newPackagesNameLoader.updateProperties({ loading: !enable });
+		await this.newPackagesVersionsLoader.updateProperties({ loading: !enable });
+		await this.newPackagesSummaryLoader.updateProperties({ loading: !enable });
+	}
+
+	private async loadNewPackageInfo(): Promise<void> {
+		await this.packageInstallButton.updateProperties({ enabled: false });
+		await this.toggleNewPackagesFields(false);
+		try {
+			let packageName = this.newPackagesSearchBar.value;
+			if (!packageName || packageName.length === 0) {
+				return;
+			}
+
+			let pipPackage = await this.fetchPypiPackage(packageName);
+			if (!pipPackage.versions || pipPackage.versions.length === 0) {
+				this.dialog.showErrorMessage(this.NoVersionsFoundError);
+				return;
+			}
+
+			await this.newPackagesName.updateProperties({
+				value: packageName
+			});
+			await this.newPackagesVersions.updateProperties({
+				values: pipPackage.versions,
+				value: pipPackage.versions[0],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: pipPackage.summary
+			});
+
+			// Only re-enable install on success
+			await this.packageInstallButton.updateProperties({ enabled: true });
+		} catch (err) {
+			this.dialog.showErrorMessage(utils.getErrorMessage(err));
+
+			await this.newPackagesName.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+			await this.newPackagesVersions.updateProperties({
+				value: this.InvalidTextPlaceholder,
+				values: [this.InvalidTextPlaceholder],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+		} finally {
+			await this.toggleNewPackagesFields(true);
+		}
+	}
+
+	private async fetchPypiPackage(packageName: string): Promise<PipPackageData> {
+		return new Promise<PipPackageData>((resolve, reject) => {
+			request.get(`https://pypi.org/pypi/${packageName}/json`, { timeout: 10000 }, (error, response, body) => {
```

Comment:
So... this won't work behind a firewall. Can you open an issue to track the need to put this into a user setting? That way they could override pypy.org with their own org URL and things will still work. No need to do that in this PR but would be a nice follow up given so many of our users have this kind of issue.

Reply:
https://github.com/microsoft/azuredatastudio/issues/5957